coffeecard/CoffeeCard.Library/Services/v2/AccountService.cs

```diff
@@ -275,5 +282,62 @@ private static string EscapeName(string name)
         {
             return name.Trim('<', '>', '{', '}');
         }
+
+        private static string GetHash(HashAlgorithm hashAlgorithm, WebhookUpdateUserGroupRequest input)
+        {
+            var sBuilder = new StringBuilder();
+            foreach (var user in input.PrivilegedUsers)
+            {
+                // Convert the input string to a byte array and compute the hash.
+                byte[] data = hashAlgorithm.ComputeHash(Encoding.UTF8.GetBytes($"{user.AccountId}:{user.UserGroup}"));
+
+                // Loop through each byte of the hashed data
+                // and format each one as a hexadecimal string.
+                for (int i = 0; i < data.Length; i++)
+                {
+                    sBuilder.Append(data[i].ToString("x2"));
+                }
+            }
+
+            // Return the hexadecimal string.
+            return sBuilder.ToString();
+        }
```

Comment:
What are your thoughts on this implementation vs overriding e.g. hashcode or equals function to calculate a deterministic hash?

Reply:
I am uncertain in which cases this would not be deterministic? 😅
Are you thinking of the scenario where the list is sent in a new order? In that case I would argue it is a new request - for one reason or another, the requester/system has actively sent a new request. We could also fix this by simply sorting the list (e.g. by id) before hashing it.

If you mean equals function such that two request objects are equal if requestA.length == requestB.length && requestA.union(requestB) == requestA.length (or a similar solution comparing the two objects contents with each other), then we would need to cache the whole request object and not just a hash value.

However please read my above comment on my take on the overall value of hashing.

Reply:
Closing this discussion